lib/utils.js

```diff
@@ -1,444 +1,491 @@
-          _hash[key] = val;
+  }
+
+  return values;
+}
+exports.mapValueFieldNames = mapValueFieldNames;
+
+function isColString(value) {
+  return typeof value === 'string' && value.substr(0, 1) === '$' && value.substr(value.length - 1, 1) === '$';
+}
+exports.isColString = isColString;
+
+function argsArePrimaryKeys(args, primaryKeys) {
+  let result = (args.length === Object.keys(primaryKeys).length);
+  if (result) {
+    _.each(args, arg => {
+      if (result) {
+        if (['number', 'string'].indexOf(typeof arg) !== -1) {
+          result = true;
+        } else {
+          result = (arg instanceof Date) || Buffer.isBuffer(arg);
         }
-      });
-
-      result = _hash;
-    }
-
-    return result;
-  },
-
-  inherit: function(SubClass, SuperClass) {
-    if (SuperClass.constructor === Function) {
-      // Normal Inheritance
-      SubClass.prototype = new SuperClass();
-      SubClass.prototype.constructor = SubClass;
-      SubClass.prototype.parent = SuperClass.prototype;
+      }
+    });
+  }
+  return result;
+}
+exports.argsArePrimaryKeys = argsArePrimaryKeys;
+
+function canTreatArrayAsAnd(arr) {
+  return arr.reduce((treatAsAnd, arg) => {
+    if (treatAsAnd) {
+      return treatAsAnd;
     } else {
-      // Pure Virtual Inheritance
-      SubClass.prototype = SuperClass;
-      SubClass.prototype.constructor = SubClass;
-      SubClass.prototype.parent = SuperClass;
+      return _.isPlainObject(arg);
     }
-
-    return SubClass;
-  },
-
-
-  stack: function _stackGrabber() {
-    var orig = Error.prepareStackTrace;
-    Error.prepareStackTrace = function(_, stack) { return stack; };
-    var err = new Error();
-    Error.captureStackTrace(err, _stackGrabber);
-    var errStack = err.stack;
-    Error.prepareStackTrace = orig;
-    return errStack;
-  },
-
-  sliceArgs: function (args, begin) {
-    begin = begin || 0;
-    var tmp = new Array(args.length - begin);
-    for (var i = begin; i < args.length; ++i) {
-        tmp[i - begin] = args[i];
+  }, false);
+}
+exports.canTreatArrayAsAnd = canTreatArrayAsAnd;
+
+function combineTableNames(tableName1, tableName2) {
+  return (tableName1.toLowerCase() < tableName2.toLowerCase()) ? (tableName1 + tableName2) : (tableName2 + tableName1);
+}
+exports.combineTableNames = combineTableNames;
+
+function singularize(s) {
+  return inflection.singularize(s);
+}
+exports.singularize = singularize;
+
+function pluralize(s) {
+  return inflection.pluralize(s);
+}
+exports.pluralize = pluralize;
+
+function removeCommentsFromFunctionString(s) {
+  s = s.replace(/\s*(\/\/.*)/g, '');
+  s = s.replace(/(\/\*[\n\r\s\S]*?\*\/)/mg, '');
+
+  return s;
+}
+exports.removeCommentsFromFunctionString = removeCommentsFromFunctionString;
+
+function toDefaultValue(value) {
+  if (typeof value === 'function') {
+    const tmp = value();
+    if (tmp instanceof DataTypes.ABSTRACT) {
```

Comment:
You can omit the `else`s here as you return in every single block, less code and easier to read imo.